ONSdigital/eq-survey-runner
# eq-635 timeout

### What is the context of this PR?

[Trello card](https://trello.com/c/Wx4tvl1f/635-timeout-facility-13)

- Added Dialog component (using [a11ydialog](http://edenspiekermann.github.io/a11y-dialog/#introduction) under the hood)
- Added Fetch component (for AJAX requests)
- Created Timeout component that displays a dialog once the threshold of the session timeout has been reached (default 120 seconds) allowing user to continue with survey or sign out.

### How to review 

- set env var `EQ_SESSION_TIMEOUT_SECONDS` to `160`
- start app
- begin a survey
- wait 10 seconds for dialog
- test 'continue survey' button resets timeout
---
- start again
- confirm 'save and sign out' button works
---
- start again
- confirm session times out
